aip/general/0203.md

```diff
@@ -16,22 +16,24 @@ behavior (for example, to optimize client library signatures).
 
 ## Guidance
 
-APIs **should** use the `google.api.field_behavior` annotation to describe
-well-understood field behavior, such as a field's being required, immutable, or
-output only:
+APIs use the `google.api.field_behavior` annotation to describe well-understood
+field behavior, such as a field being required or immutable.
 
 ```proto
 // The audio data to be recognized.
 RecognitionAudio audio = 2 [(google.api.field_behavior) = REQUIRED];
 ```
 
-Additionally, APIs **may** use the `OPTIONAL` value to describe none of the
-above. However, it is never mandatory to explicitly describe a field as
-optional.
+- APIs **must** use the `google.api.field_behavior` annotation on every field.
+- If an option described in this AIP accurately describes the behavior of the
+  field, it **must** be annotated on the field.
```

Comment:
I'm still struggling with this part of the guidance. My concerns are the following:

* Having "if" and **must** confuses the issue.
* "an option described in the AIP" is vague because I don't knew the exact set you are referring too; values are discussed throughout. I would either,
  * explicitly say something like, "The value of the annotation **must** be one of the those described in the`[Vocabulary](#vocabulary)` section of this AIP".
  * or, link to the enum with the values using similar verbiage as that provided above, https://github.com/googleapis/googleapis/blob/master/google/api/field_behavior.proto

The advantage to linking to the proto is that when values are added, this is capture. For example, "Vocabulary" does not cover, `NON_EMPTY_DEFAULT`. If you intend to go with Option 1 above, I would suggest adding a section for `NONE_EMPTY_DEFAULT`. In fact, we should probably do this regardless.

If you do link to the file in googleapis/googleapis, again, I would be explicit as possible, e.g. make sure it is clear that the `FIELD_BEHAVIOR_UNSPECIFIED` value **must not** be used.

Reply:
I definitely see your point. My intention was to force someone to read through
these annotations to see what is relevant and then apply them. 

I like your idea about linking to the proto, I'll do that. How about I just add
a warning here that says "if you add annotations afterward, they could cause
backward-incompatible changes in the clients. Therefore, if it's important to
review the relevant annotations and apply them when the field is introduced.".

or something like that. WDYT?